internal/zinc-core/src/main/scala/sbt/internal/inc/IncrementalCommon.scala

```diff
@@ -359,17 +352,18 @@ private[inc] abstract class IncrementalCommon(log: sbt.util.Logger, options: Inc
               inv("stamp changed from " + previousStamp + " to " + currentStamp)
           }
         def dependencyModified(file: File): Boolean =
-          previous.className(file) match {
-            case None => inv("no class name was mapped for it.")
-            case Some(name) => lookup.lookupOnClasspath(name) match {
-              case None    => inv("could not find class " + name + " on the classpath.")
-              case Some(e) => entryModified(name, e)
+          {
+            val classNames = previousRelations.libraryClassNames(file)
+            if (classNames.isEmpty) false
+            else classNames exists { binaryClassName =>
+              lookup.lookupAnalysis(binaryClassName) match {
```

Comment:
What with shadowing by other library? In theory it should work in previous implementation?